### PR TITLE
Mbarba/brc mode

### PR DESCRIPTION
**Pull request by MatBarba**
Clean up brc_mode usage:

- define as 0 by default in all pipelines
- remove from scripts where it is not used
- use params.brc_mode universally
- add some basic pipeline tests

### CODE REVIEW

File: pipelines/nextflow/workflows/dumper_pipeline/main.nf [hunk 1]
```diff
@@ -34,13 +33,13 @@ default_selection = [
...
         Mandatory arguments:
-        --host, --port, --user           Connection parameters to the SQL servers we getting core db(s) from
+        --host, --port, --user         Connection parameters to the SQL servers we getting core db(s) from
 
         Optional arguments:
         --password                     Password part of the connection parameters
         --prefix                       Core dabase(s) name prefixes
         --dbname_re                    Regexp to match core db name(s) against
-        --brc_mode	               Override Ensembl 'species' and 'division' with the corresponding BRC4 ones ('organism_abbrev' and 'component')
+        --brc_mode	                   Override Ensembl 'species' and 'division' with the corresponding BRC4 ones ('organism_abbrev' and 'component')
```
Comment (JAlvarezJarreta) [inline pipelines/nextflow/workflows/dumper_pipeline/main.nf:42]:
```suggestion
        --brc_mode	               Override Ensembl 'species' and 'division' with the corresponding BRC4 ones ('organism_abbrev' and 'component')
```

File: pipelines/nextflow/workflows/additional_seq_prepare/main.nf [hunk 2]
```diff
@@ -38,7 +37,7 @@ def helpMessage() {
         --output_dir                   Output directory to place final output
         --cache_dir                    Cache directory for downloaded files
         --help                         This usage statement.
-        --brc_mode                     By default it is set to 1, set it to 0 if you are not using it for brc
+        --brc_mode                     Set to 1 to use with BRC data (default: ${paranms.brc_mode})
```
Comment (JAlvarezJarreta) [inline pipelines/nextflow/workflows/additional_seq_prepare/main.nf:40]:
```suggestion
        --brc_mode                     Set to 1 to use with BRC data (default: ${params.brc_mode})
```

**Review by JAlvarezJarreta — CHANGES_REQUESTED** [review]
Looks great, couple of minor changes.

Two comments (not affecting the approval in any way):
* I fear how this is going to be merged with PR#173 (there are places that have been changed in both)
* I'm wondering if there is something similar than the parametrize in pytest for the Nextflow tests to avoid reflecting information such as `_brc_mode_of` in the name...

**Comment by MatBarba** [comment]
> Looks great, couple of minor changes.
> 
> Two comments (not affecting the approval in any way):
> 
>     * I fear how this is going to be merged with PR#173 (there are places that have been changed in both)
Rebase/merges make it pretty easy to resolve conflicts along the way. We just need to make sure that the tests work in the end.
> 
>     * I'm wondering if there is something similar than the parametrize in pytest for the Nextflow tests to avoid reflecting information such as `_brc_mode_of` in the name...
Some of the files contents should actually differ, so we do need that distinction. I don't intend to add other variations though.

**Review by JAlvarezJarreta — APPROVED** [review]

**Pull request merged**